docs/docs/node-interface.md

```diff
@@ -11,11 +11,18 @@ The basic node data structure is as follows:
 id: String,
 children: Array[String],
 parent: String,
+// Reserved for plugins who wish to extend other nodes.
+pluginFields: Object,
 internal: {
   contentDigest: String,
   mediaType: String,
+  // A globally unique node type choosen by the plugin owner.
   type: String,
-  pluginName: String,
+  // The plugin which created this node.
+  pluginOwner: String,
+  // Stores which plugins created which fields.
+  fieldPluginOwners: Object,
```

Comment:
Also, thinking about it more, `fields` would be a bit of a weird name within Gatsby for a plugin to use when creating nodes as either you just put whatever is under `fields` as a top-level value or they should be children nodes.

Another argument for `pluginFields` however is that people will be confused as to why some data is in top-level fields and other is under `fields`. `pluginFields` would make it sorta obvious what the difference is. Though... perhaps not obvious enough to counter the extra length.

Most people doing anything with Gatsby will soon learn the distinction as adding custom fields to nodes is something many people will need to do for anything of medium or higher complexity...

Other thoughts? @jquense @fabien0102 @fk @scottyeck @nicholaswyoung